### ISSUE TITLE
Transforms causes game to freeze upon launch

### ISSUE THREAD

**Issue by SuperHatGuy** [renpy/renpy#6014]
## Device info
**OS:** Linux, Kubuntu 24.04 LTS
**Vendor:** AMD
**Version:** Ren'Py 8.3.4.24120703 stable

## Bug description
Upon launching the game, the main menu appears and the game may briefly work normally before the application freezes and stops completely.
No errors, exceptions, nor logs are produced. Removing or commenting any element of the script fixes the freezing.

## Code to reproduce
```py
# script.rpy
image eileen = Placeholder("girl")

transform first(child):
    child
    child

transform second(child):
    child

label start:
    show eileen at first, second
    return
```

**Comment by danac78** [comment]
Your reproduce code seems to be missing something based on:

[limitreach](https://www.renpy.org/doc/html/transforms.html#atl-curry-and-partial-parameter-passing). Because if you do it without the child mentions, it does both transform in that list.

**Comment by renpytom** [comment]
Yeah, child is special now, and setting the child to the child is certainly confusing Ren'Py. I'd suggest just renaming the variable to something else.

**Comment by SuperHatGuy** [comment]
@danac78 Both transforms are using the special `child` parameter, which is given automatically when the transforms execute.
The reproduce code is meant to be a boiled down version of the code that provokes the freeze.
Curiously, removing anything else pretty much stops the freeze from happening. I'd like to add that the problem persists even when replacing the parameter with `child=None` in both transform declarations.

**Comment by SuperHatGuy** [comment]
@renpytom If it's relevant, this is the context in which I ran into this bug:
I was experimenting with transforms, and my goal was to implement fading for character layeredimages, as well as zooming the current speaker automatically.
In my project, the `first` transform would fade `child` in and out using transitions, and the `second` transform uses `child` to check its attributes for the speaker attribute and zooming it accordingly.
That's how I found the bug unexpectedly. Considering it produced no errors, I decided to open this issue.
